test/unit/gateways/safe_charge_test.rb

```diff
@@ -225,6 +235,34 @@ def test_3ds_response
     assert_equal 'https://pit.3dsecure.net/VbVTestSuiteService/pit1/acsService/paReq?summary=MjRlZGYwY2EtZTk5Zi00NDJjLTljOTAtNWUxZmRhMjEwODg3', purchase.params['acsurl']
   end
 
+  def test_mpi_response_fail
+    purchase = stub_comms do
+      @gateway.purchase(@amount, @three_ds_enrolled_card, @mpi_options)
+    end.check_request do |endpoint, data, headers|
```

Comment:
Rubocop failed for me due to unused parameters not having prefixed underscores here and on L255.